### PR TITLE
Set DISPLAY environment variable for browser on Linux

### PR DESCRIPTION
**Pull request by mtmiller**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This change stashes the user's original `DISPLAY` into `HOMEBREW_DISPLAY` and restores it when calling out to the browser. See #5692.

I suspect this should also be documented in docs and/or manpages, right? Should I also add a test case, something like `HOMEBREW_BROWSER='echo $DISPLAY'? Any other ways I can make this better?

Cc @sjackman for review/advice

### CODE REVIEW

File: Library/Homebrew/utils.rb [hunk 1]
```diff
@@ -388,6 +388,10 @@ def exec_browser(*args)
   browser ||= OS::PATH_OPEN if defined?(OS::PATH_OPEN)
   return unless browser
 
+  if ENV["HOMEBREW_DISPLAY"]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils.rb]: Can skip the `if`; won't do any harm if it's not set.

**Comment by MikeMcQuaid** [comment]
> I suspect this should also be documented in docs and/or manpages, right? 

Yes, please! Under "environment variables" would be ideal.

> Should I also add a test case, something like `HOMEBREW_BROWSER='echo $DISPLAY'? Any other ways I can make this better?

Don't need a test for this, thanks!

File: Library/Homebrew/utils.rb (hunk 1 above)
Comment (sjackman) [reply Library/Homebrew/utils.rb]: My personal preference would be…
```ruby
ENV["DISPLAY"] = ENV["HOMEBREW_DISPLAY"] if ENV["HOMEBREW_DISPLAY"]
```
The error message from a browser may be different if the `DISPLAY` environment variable is not set versus set to the empty string.
@mtmiller Could you please test…
```sh
env -u DISPLAY firefox
env DISPLAY= firefox
```
and see what the error message is?

Comment (mtmiller) [reply Library/Homebrew/utils.rb]: Thanks for the feedback. I eliminated the `if` entirely. I'm not much of a rubyist, but I found that `ENV` DTRT, assigning `nil` deletes from the environment.

Comment (mtmiller) [reply Library/Homebrew/utils.rb]: > Could you please test…
> and see what the error message is?

I did, the error message does differ by browser, but the effect is the same, no tab opened.

Firefox does differentiate:

```
$ env -u DISPLAY firefox
Error: no DISPLAY environment variable specified
$ env DISPLAY= firefox
Unable to init server: Broadway display type not supported: 
Error: cannot open display: 
```

Chrome doesn't care whether it's unset or empty:

```
$ env -u DISPLAY google-chrome

(google-chrome:32416): Gtk-WARNING **: 12:48:05.569: cannot open display: 
$ env DISPLAY= google-chrome

(google-chrome:32480): Gtk-WARNING **: 12:48:09.083: cannot open display: 
```

**Comment by mtmiller** [comment]
> Yes, please! Under "environment variables" would be ideal.

Done.

**Comment by mtmiller** [comment]
Thanks for the guidance and feedback to get this done.

This is my first contribution to Homebrew, and I'm not a rubyist, so it took a little while to get past some bundle/gem setup problems to generate the man pages, but got it working in the end.

Happy to squash commits, rewrite commit messages, or add anything else you want to see in this PR.

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @mtmiller!

**Pull request merged**